sim/mage/items.go

```diff
@@ -83,11 +84,47 @@ var ItemSetTimeLordsRegalia = core.NewItemSet(core.ItemSet{
 		// Your Arcane Blast has a 100% chance and your Fireball, Pyroblast, Frostfire Bolt, and Frostbolt spells have a 50% chance to grant Stolen Time, increasing your haste rating by 50 for 30 sec and stacking up to 10 times.
 		// When Arcane Power, Combustion, or Icy Veins expires, all stacks of Stolen Time are lost.
 		2: func(agent core.Agent) {
-			// mage := agent.(MageAgent).GetMage()
+			character := agent.GetCharacter()
+			mage := agent.(MageAgent).GetMage()
+
+			// Stack reset handlers can be found in:
+			// combustion.go
+			// talents_arcane.go
+			// talents_frost.go
+			mage.t13ProcAura = core.MakeStackingAura(character, core.StackingStatAura{
+				Aura: core.Aura{
+					Label:     "Stolen Time",
+					ActionID:  core.ActionID{SpellID: 105785},
+					Duration:  time.Second * 30,
+					MaxStacks: 10,
+				},
+				BonusPerStack: stats.Stats{stats.HasteRating: 50},
+			})
+
+			newStolenTimeTrigger := func(procChance float64, spellMask int64) {
+				core.MakePermanent(core.MakeProcTriggerAura(&character.Unit, core.ProcTrigger{
```

Comment:
You don't need the `MakePermanent()` wrapper here, that's done by default if no `Duration:` is supplied in the `ProcTrigger` config.